switch.py

```diff
@@ -34,17 +34,95 @@ async def async_setup_entry(
         ]
     if router.tailscale_configured:
         switches.append(TailscaleSwitch(router))
+    for iface in router._wifi_ifaces:
+        switches.append(WifiApSwitch(router, iface))
     if switches:
         async_add_entities(switches, True)
 
 
-class TailscaleSwitch(SwitchEntity):
-    """A tailscale switch."""
+class GliSwitchBase(SwitchEntity):
+    """GL-inet switch base class."""
 
     def __init__(self, router: GLinetRouter) -> None:
         """Initialize a GLinet device."""
         self._router = router
-        # self._client = client
+
+    _attr_has_entity_name = True
+
+    @property
+    def entity_category(self) -> EntityCategory:
+        """A config entity."""
+        return EntityCategory.CONFIG
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information."""
+        # TODO this should probably be defined in the router device not here in the switch
+        data: DeviceInfo = {
+            "connections": {(CONNECTION_NETWORK_MAC, self._router.factory_mac)},
+            "identifiers": {(DOMAIN, self._router.factory_mac)},
+        }
+        return data
+
+
+class WifiApSwitch(GliSwitchBase):
+    """A WiFi AccessPoint switch."""
+
+    def __init__(self, router: GLinetRouter, iface_name: str) -> None:
+        """Initialize a GLinet device."""
+        super().__init__(router)
+        self._iface_name = iface_name
+
+    @property
+    def icon(self) -> str:
+        """Return AP state icon."""
+        if self.is_on:
+            return "mdi:wifi"
+        return "mdi:wifi-off"
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return self._iface_name
```

Comment:
```suggestion
        return self._iface.get('ssid', self._iface_name)
```
results in prettier names (that are not necessarily unqiue if a user has 2.4 and 5Ghz as the same)